chain/client/src/sync/state/downloader.rs

```diff
@@ -156,7 +156,7 @@ impl StateSyncDownloader {
             let i = AtomicUsize::new(0); // for easier Rust async capture
             let attempt = || async {
                 let source = if i.load(Ordering::Relaxed) < num_attempts_before_fallback
-                    && fallback_source.is_some()
+                    || fallback_source.is_none()
```

Comment:
I'm so sorry, I fixed one place but not the other.

Would you mind making this consistent with line 72? I'm fine with either approach.

Reply:
Oh I see, yes, I'll make them consistent. Thanks